src/poly.pest

```diff
@@ -2,13 +2,13 @@ name = { (!"\n" ~ ANY)+ }
 
 separator = _{ (" " | "\t")+ }
 
-decimal = _{ sign? ~ number ~ ("." ~ number ~ ("E" ~ sign ~ number)?)? }
+decimal = _{ sign? ~ number ~ ("." ~ number)? ~ (("E" | "e") ~ sign? ~ number)? }
 sign = _{ "+" | "-" }
 number = _{ ASCII_DIGIT+ }
 
-file = { SOI ~ (ring ~ NEWLINE)* ~ "END" ~ NEWLINE* ~ EOI }
+file = { SOI ~ name ~ NEWLINE ~ (ring ~ NEWLINE)* ~ "END" ~ NEWLINE* ~ EOI }
 
-ring = { (name ~ NEWLINE)? ~ subtract? ~ name ~ NEWLINE ~ (point ~ NEWLINE)* ~ "END" }
+ring = { (name | (subtract ~ name)) ~ NEWLINE ~ (point ~ NEWLINE)* ~ "END" }
```

Comment:
here was the main problem: this swallowed the first point of the second polygon if that second polygon was not a subtraction.

I moved the first-name-parsing to the file level. I hope I did not break another case, but it works with all the poly files from geofabrik now.